### PR TITLE
IPFS Integration

### LINKED ISSUE #25: UI should wait for ChluIPFS to be ready
When switching mode or when starting the app, the UI should wait for ChluIPFS to be ready

### PR DESCRIPTION
**Pull request by fazo96**
__WIP, not done yet__

What's included:

- UI waits for ChluIPFS to be ready (closes #25)
- UI switches ChluIPFS node type when switching between vendor and customer
- ChluIPFS is called to create a review when the pay button is pressed, part of #24 

What's missing:

- ChluIPFS dependency is not using master branch
- while setting up ChluIPFS, the page is blank, should it show a loading spinner?
- some tests fail due to async/await being used in chlu-demo. In other parts of the code there are other async/await calls but they work fine, I need to investigate. Outside of the tests everything works

### CODE REVIEW

File: customer-wallet/src/routes.js [hunk 1]
```diff
@@ -64,20 +65,28 @@ function checkMnemonicExists(proceed) {
...
-    dispatch(getProfile(nextUserType))
-      .then(({ userType }) => {
-        const nextPath = [ nextUserType || userType, ...rest ].join('/')
-        replace(`/${nextPath}`)
-      })
-      .catch(error => {
-        console.log('error_____', error)
-        proceed()
-      })
+    try {
```
Comment (kulpreet) [inline customer-wallet/src/routes.js]: @Venumteam - You OK with using async/await here?
Actually @fazo96 why the change to async await here?

File: customer-wallet/src/containers/Customer/CustomerWallet/CustomerWalletForm/index.js [hunk 2]
```diff
@@ -47,12 +48,15 @@ class CustomerWalletFormWrapper extends Component {
   handleChangeAddress = (activeAddress) => this.setState({ activeAddress })
 
   handleSubmit = ({ toAddress, amountBtc }) => {
-    this.setState({ isDisabledSubmit: true }, () => {
-      const { rating, convertBitsToSatoshi } = this.props
+    this.setState({ isDisabledSubmit: true }, async () => {
+      const { rating, convertBTCToSatoshi } = this.props
```
Comment (kulpreet) [inline customer-wallet/src/containers/Customer/CustomerWallet/CustomerWalletForm/index.js]: @fazo96 I think there is an error here, it should be convertBitsToSatoshi.
@Venumteam - can you confirm?

File: customer-wallet/src/containers/Customer/CustomerWallet/CustomerWalletForm/index.js [hunk 3]
```diff
@@ -47,12 +48,15 @@ class CustomerWalletFormWrapper extends Component {
...
       const { blockchainClient: { createChluTransaction: tr } } = this.context
       const { activeAddress } = this.state
 
-      tr.create(activeAddress, toAddress, round(convertBitsToSatoshi(parseFloat(amountBtc))), null, 'Hello World')
+      const chluIpfs = await getChluIPFS(types.customer)
+      const contentHash = await chluIpfs.storeReviewRecord(Buffer.from('fake review record content'))
+
+      tr.create(activeAddress, toAddress, convertBTCToSatoshi(parseFloat(amountBtc)), null, contentHash)
```
Comment (kulpreet) [inline customer-wallet/src/containers/Customer/CustomerWallet/CustomerWalletForm/index.js]: @fazo96 same as above, I think this should be convertBitsToSatoshi

File: customer-wallet/src/containers/Customer/CustomerWallet/CustomerWalletForm/index.js (hunk 2 above)
Comment (Venumteam) [reply customer-wallet/src/containers/Customer/CustomerWallet/CustomerWalletForm/index.js]: @kulpreet you are right, there should be `convertBitsToSatoshi` which we already get from input form

Comment (fazo96) [reply customer-wallet/src/containers/Customer/CustomerWallet/CustomerWalletForm/index.js]: Sorry guys, I made a mistake when rebasing

File: customer-wallet/src/routes.js (hunk 1 above)
Comment (fazo96) [reply customer-wallet/src/routes.js]: I changed it to async/await since it made the code a little more readable and I saw async/await being used in other parts of the code. Let me know if you prefer to revert to promises

Comment (kulpreet) [reply customer-wallet/src/routes.js]: @fazo96 async/await is ok. I just wanted Max to confirm this, and he did, so we are good with this change.

File: customer-wallet/src/containers/Customer/CustomerWallet/CustomerWalletForm/index.js (hunk 2 above)
Comment (kulpreet) [reply customer-wallet/src/containers/Customer/CustomerWallet/CustomerWalletForm/index.js]: @fazo96 that's ok. Can you push a fix for this and then we can merge this.
